test/BlockStrings.hs

```diff
@@ -0,0 +1,42 @@
+
+-- | Regression tests for issue #20 https://github.com/hasura/graphql-parser-hs/issues/20
+
+module BlockStrings where
+
+import           Hedgehog
+import           Language.GraphQL.Draft.Parser
+import qualified Data.Text as T
+import           Data.Attoparsec.Text
+
+blockTest :: IO Bool
+blockTest = do
+  checkParallel $ Group "Test.parser.block-string.unit"
+    [ ("parses the specExample", blockParsesTo "\n    Hello,\n      World!\n\n    Yours,\n      GraphQL.\n  " "Hello,\n  World!\n\nYours,\n  GraphQL.")
+    , ("parses empty string", blockParsesTo "" "")
+    , ("parses newline", blockParsesTo "\n" "")
+    , ("parses newlines in empty block", blockParsesTo " \n   \n  \n            \n\n " "")
```

Comment:
I don't think this is what the [specification](http://spec.graphql.org/June2018/#BlockStringValue()) says. I think `BlockStringValue("\n \n \n \n\n ")` would compute to be `" \n \n \n"`.

Reply:
So we use the indentation of the last line to determine the smallest indentation, which is 0, and so do not remove the indentation of the previous lines even though the last line will be removed in the end of the reformatting? Is that the correct understanding? This is a weird one.

Reply:
Looking at this again, I'm not sure I interpreted the spec correctly. Perhaps this test is correct after all. `BlockStringValue` is an odd beast.

Reply:
Yeah, I'm having some trouble with this one, but you are correct about removing only first and last lines anyway.

Reply:
We both got confused.
As Lorenzo pointed out to me in the spec for BlockStringValue we remove all empty lines case there is only empty lines, because the step 5 includes a **while** that we missed.
So I going to guide the implementation in this direction.